### DISCUSSION TITLE
Annoying Tooltips that cant be disabled

### DISCUSSION THREAD

**Discussion by escapingSamsara** [LazyVim/LazyVim#2885] [Q&A] [answered]
I am currently facing an issue that was most probably  introduced with one of the latest updates to the lazyvim distro. 

I constantly see these (MDN references) when writing my code (screenshot showing when writing "console.log("), and i also cant remove or toggle the signature helpers, they simply always show up (on the completion menu/dropdown). is there a way to fix this because having so many popups is driving me nuts.

**Comment by folke** [comment]
I have no idea what you're talking about. Nothing changed in Lazyvim that might be related

**Comment by briandipalma** [comment]
I can't find your dot files repo so can't see what config you have. Also this is the wrong repo to open an issue like that. You could have changed the config for https://github.com/folke/noice.nvim or https://github.com/hrsh7th/nvim-cmp and that could be causing an issue. Have you tried disabling them?

**Comment by dpetka2001** [comment, marked as answer]
In your `noice.nvim` spec you can set `lsp.signature.auto_open = false` and then only manually call for the LSP signature when you need via the keymap `<C-k>` in `insert` mode. You can also pass to `opts` settings like `max_height` etc to more granually specify how big the popup window should get.

**Reply by jni** [reply]
Hi folks, apologies for the n00b question, I am brand new at LazyVim/neovim (installed a couple of hours ago). I also want to disable the auto-open of the completion tooltips, but I literally don't know what to write in which file. ie what/where is my "noice.nvim spec"? 🙏 Thanks!

**Reply by dpetka2001** [reply]
Create a file `~/config/nvim/lua/plugins/noice.lua` with following contents
```lua
-- ~/.config/nvim/lua/plugins/noice.lua
return {
{
  "folke/noice.nvim",
  opts = {
    lsp = {
      signature = {
        auto_open = {
          enabled = false,
        },
      },
    },
  },
},
}
```

**Reply by jni** [reply]
Ach, thank you for the lightning quick response, but that didn't help unfortunately!

https://github.com/LazyVim/LazyVim/assets/492549/3d6b6fad-5091-4159-a484-85a67f6229e3

The text of that file is copy-pasted from yours:

```sh
$ cat ~/.config/nvim/lua/plugins/noice.lua
return {
  {
    "folke/noice.nvim",
    opts = {
      lsp = {
        signature = {
          auto_open = {
            enabled = false,
          },
        },
      },
    },
  },
}
```

Could it be nvim-cmp that's responsible? (And how would I tell?)

Thanks again! 🙏

**Reply by folke** [reply]
Yes. that's just auto completion. If you dont want that, disable cmp

**Reply by jni** [reply]
(a) I would like to have auto-completion, but I would like it to happen on-demand with a key-binding (say tab, but I am flexible), not automatically-as-I-type. And (b) I actually don't know how to disable individual plugins, see "n00b" above. 😅 

Thanks! 🙏

**Reply by dpetka2001** [reply]
You can do
```lua
  {
    "nvim-cmp",
    opts = function(_, opts)
      opts.completion = {
        autocomplete = false,
      }
    end,
  },
```
This way you can just press `<C-Space>` (default LazyVim keymap) to manually invoke auto-completion.

**Reply by jni** [reply]
It worked! 🚀 🥳 I had to use my super-advanced pattern-matching skills to deduce that I had to put

```lua
return {
  {
    "nvim-cmp",
    opts = function(_, opts)
      opts.completion = {
        autocomplete = false,
      }
    end,
  },
}
```

in `~/.config/nvim/lua/plugins/nvim-cmp.lua`, ...

But it worked! 🥳 Thank you both for your help here! 🙏
